File: AssetSync/src/AssetSync.Core/Models/LogEntry.cs
using AssetSync.Core.Enums;

namespace AssetSync.Core.Models;

public class LogEntry
{
    public long Id { get; set; }
    public DateTimeOffset TimestampUtc { get; set; }
    public LogLevel Level { get; set; }
    public SourceSystem SourceSystem { get; set; }
    public string Action { get; set; } = string.Empty;
    public string? SerialNumber { get; set; }
    public string? DeviceName { get; set; }
    public string? FieldName { get; set; }
    public string? OldValue { get; set; }
    public string? NewValue { get; set; }
    public bool Success { get; set; }
    public string? ErrorDetail { get; set; }
    public string SyncRunId { get; set; } = string.Empty;
}

File: AssetSync-macOS/AssetSyncCore/Sources/AssetSyncCore/Models/LogEntry.swift
import Foundation

public struct LogEntry: Identifiable, Sendable {
    public var id: Int64 = 0
    public var timestampUtc: Date = Date()
    public var level: LogLevel = .info
    public var sourceSystem: SourceSystem = .application
    public var action: String = ""
    public var serialNumber: String?
    public var deviceName: String?
    public var fieldName: String?
    public var oldValue: String?
    public var newValue: String?
    public var success: Bool = true
    public var errorDetail: String?
    public var syncRunId: String = ""

    public init() {}
}

public struct LogFilter: Sendable {
    public var fromUtc: Date?
    public var toUtc: Date?
    public var minLevel: LogLevel?
    public var sourceSystem: SourceSystem?
    public var action: String?
    public var serialNumber: String?
    public var freeText: String?
    public var syncRunId: String?
    public var limit: Int?
    public var offset: Int?

    public init() {}
}
